index.js

```diff
@@ -225,10 +231,9 @@ function logger(options) {
 
             if(options.meta !== false) {
               var logData = {};
-              var bodyWhitelist, blacklist;
 
-              requestWhitelist = requestWhitelist.concat(req._routeWhitelists.req || []);
-              responseWhitelist = responseWhitelist.concat(req._routeWhitelists.res || []);
+              var requestWhitelist = options.requestWhitelist.concat(req._routeWhitelists.req || []);
+              var responseWhitelist = options.responseWhitelist.concat(req._routeWhitelists.res || []);
 
```

Comment:
I think you might have actually fixed an (unreported) bug here!

In the old code, it looks like the core `requestWhitelist` and `responseWhitelist` were combined with the corresponding route whitelists. And because of the way the whitelists were exported, the exported array would still be the original default whitelist while the module internally would be checking the combined arrays. And those combined arrays would presumably be getting larger and larger with each route containing its own route whitelists.

So thanks! :smile: